### ISSUE TITLE
Detekt isn't able to detect issues on iosMain source set

### ISSUE THREAD

**Issue by tfcporciuncula** [detekt/detekt#5609]
## Expected Behavior
Detekt should be able to detekt issues on the `iosMain` sourceset in a KMM project.

## Observed Behavior
I've tried several different tasks and none of them are able to do it, including:
- `detektMetadataIosMain`
- `detektIosX64Main`
- `detektIosArm64Main`
- `detektIosSimulatorArm64Main`

And `detektMetadataMain` only finds issues on `commonMain`.

## Steps to Reproduce
- Clone https://github.com/tfcporciuncula/detekt-kmm-issue
- Observe there's [an issue in the iOS source set of the shared module](https://github.com/tfcporciuncula/detekt-kmm-issue/blob/534d3c5dda536c9ad5aedf4dd698d68dc8046768/shared/src/iosMain/kotlin/com/fredporciuncula/detekt/Platform.kt#L7)
- Run any Detekt task and observe the issue isn't detected
- A summary of what happens can be observed in this commit: https://github.com/tfcporciuncula/detekt-kmm-issue/commit/54f2fca894ced9f20134cdc6c45e050e80f28e34

## Context
I'm just trying to run Detekt in a multiplatform project. Even though the documentation isn't clear about it, it seems the way to do it is to run multiple tasks for the different source sets, but unfortunately I haven't found a task able to find issues on code inside `iosMain`.

Currently, I'm actually using a workaround to set the module root as the source set for Detekt, so all source sets are analyzed:
```kotlin
tasks.withType<io.gitlab.arturbosch.detekt.Detekt> detekt@{
  setSource(files(project.projectDir))
  exclude("**/*.kts")
  exclude("**/build/**")
}
```

And then I can just run the `detekt` task and make sure it's covering the entire project (with no type resolution).

## Your Environment
* Version of detekt used: 1.22.0
* Version of Gradle used (if applicable): 7.4 and 
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: macOS Monterey 12.6
* Link to your project (if it's a public repository): https://github.com/tfcporciuncula/detekt-kmm-issue

**Comment by cortinico** [comment]
Thanks for reporting this @tfcporciuncula, and thanks for sending a reproducer.

We add KMP support last year and it was experimental as we couldn't find much user interested in it. Thanks for getting back to us so we can investigate and fix this accordingly 👍 

I've briefly looked into the reproducer and it looks fine at a first glance, we'll have to investigate further. The only thing to note also is that iOS/Native/JS targets don't support Type Resolution rules. I see you're testing with a rule that does not use Type Resolution, so that should definitely work.

**Comment by mustafaozhan** [comment]
Up 🆙

**Comment by 3flex** [comment]
This is fixed in detekt 2.0, most likely by https://github.com/detekt/detekt/pull/6973
